test-alerts.cpp

```diff
@@ -5,4 +5,35 @@
 
 TEST_CASE("infers the breach according to limits") {
   REQUIRE(inferBreach(12, 20, 30) == TOO_LOW);
+  REQUIRE(inferBreach(30, 20, 30) == NORMAL);
+  REQUIRE(inferBreach(31, 20, 30) == TOO_HIGH);
+  REQUIRE(inferBreach(20, 20, 30) == NORMAL);
+}
+
+TEST_CASE("classify breach")
+{
+    REQUIRE(classifyTemperatureBreach(PASSIVE_COOLING, -1) == TOO_LOW);
+    REQUIRE(classifyTemperatureBreach(MED_ACTIVE_COOLING, -1) == TOO_LOW);
+
+    REQUIRE(classifyTemperatureBreach(PASSIVE_COOLING, 0) == NORMAL);
+    REQUIRE(classifyTemperatureBreach(PASSIVE_COOLING, 20) == NORMAL);
+    REQUIRE(classifyTemperatureBreach(PASSIVE_COOLING, 35) == NORMAL);
+    REQUIRE(classifyTemperatureBreach(MED_ACTIVE_COOLING, 0) == NORMAL);
+    REQUIRE(classifyTemperatureBreach(MED_ACTIVE_COOLING, 20) == NORMAL);
+    REQUIRE(classifyTemperatureBreach(MED_ACTIVE_COOLING, 40) == NORMAL);
+
+    REQUIRE(classifyTemperatureBreach(PASSIVE_COOLING, 36) == TOO_HIGH);
+    REQUIRE(classifyTemperatureBreach(MED_ACTIVE_COOLING, 41) == TOO_HIGH);
+}
+
+TEST_CASE("check classify and alert")
+{
+    BatteryCharacter batteryAttribute = {PASSIVE_COOLING, "LUMINOUS"};
+    checkAndAlert(batteryAttribute, 36, sendToController);
```

Comment:
This method does not have asserts.
The breach messages received and the recipient needs to be verified, until then the tests are weak in nature...